src/middlewares/is_signer_ready.rs

```diff
@@ -0,0 +1,17 @@
+use crate::hooks::{use_accounts::UseAccountsState, use_notification::UseNotificationState};
+use dioxus_std::{i18n::UseI18, translate};
+pub fn is_signer_ready(
+    i18: UseI18,
+    accounts: UseAccountsState,
+    mut notification: UseNotificationState,
+) -> impl FnOnce() -> Result<(), &'static str> {
+    move || {
+        if accounts.get_account().is_none() {
+            notification.handle_warning(&translate!(i18, "warnings.middleware.has_dao"));
+            Err("Failed to get account to sign")
```

Comment:
How is this error going to be handled? is it going to be logged or shown to the user(then it would need i18n)?

Reply:
The notification uses i18n and displays a warning to the user. The error is just a resource for development.